clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -111,12 +126,401 @@ class image_mem_wrapper {
       const sycl::ext::oneapi::experimental::image_mem_handle &handle)
       : _channel(channel), _desc(desc), _handle(handle) {}
 
+  /// Initialize image mem wrappers for all the levels of mipmap image
+  /// \param [in] q The Queue to be used to query the mip levels
+  void init_mip_level_wrappers(sycl::queue q) {
+    auto num_levels = _desc.num_levels;
+    if (_desc.type == sycl::ext::oneapi::experimental::image_type::mipmap) {
+      assert(num_levels > 1);
+      _sub_wrappers = (image_mem_wrapper *)std::malloc(
+          sizeof(image_mem_wrapper) * num_levels);
+      for (unsigned i = 0; i < num_levels; ++i)
+        new (_sub_wrappers + i) image_mem_wrapper(
+            _channel, _desc.get_mip_level_desc(i),
+            sycl::ext::oneapi::experimental::get_mip_level_mem_handle(
+                _handle, i, q.get_device(), q.get_context()));
+    }
+  }
+
   image_channel _channel;
   const sycl::ext::oneapi::experimental::image_descriptor _desc;
   sycl::ext::oneapi::experimental::image_mem_handle _handle;
   image_mem_wrapper *_sub_wrappers{nullptr};
 };
 
+/// The base wrapper class of graphics interop memory handle.
+class interop_mem_wrapper_base {
+public:
+  /// Creates interop memory wrapper for a given resource.
+  interop_mem_wrapper_base() = default;
```

Comment:
```suggestion
  interop_mem_wrapper_base(sycl::ext::oneapi::experimental::interop_mem_handle interop_mem_handle); // Initialize the interop_mem_handl
```

Put it to `protected:`